useeffect-and-calling-apis/useeffect-after-render.md

```diff
@@ -85,32 +53,29 @@ Targeted questions will bring us back to this topic when we need to. Imagine deb
 
 ## `useEffect` Executes After Render
 
-Different [React hooks](https://reactjs.org/docs/hooks-intro.html) allow us to "hook" into specific parts of the lifecycle.
-
 The [**`useEffect` hook**](https://reactjs.org/docs/hooks-effect.html) is a hook function that runs during these lifecycle steps:
 
-- `componentDidMount`
-- `componentDidUpdate`
-- `componentWillUnmount`
+- `Mounting Stage`
+- `Updating Stage` - When specific props or state variables change
+- `Unmounting Stage`
 
-This means that the `useEffect` hooks runs:
+This means that the `useEffect` hooks can run:
 
-- once in the Mounting stage, after the component is successfully inserted into the DOM and fully renders
-- once in the Update stage, after the component fully updates and re-renders
-- once in the Unmounting stage, right before the component is removed from the DOM
-
-There is an important pattern here: both `componentDidMount` and `componentDidUpdate` occur **_after_** a component fully renders.
+- Once in the `Mounting Stage`, *after* the component is successfully inserted into the DOM and fully renders
+- In the `Update Stage`, after a prop or state variable changes
+- Once in the `Unmounting Stage`, right before the component is removed from the DOM
 
 In summary, we can say that `useEffect` is called after a component renders.
 
 ## Calling `useEffect`
 
 To use the `useEffect` hook, we call the `useEffect` function with two parameters:
 
-1. A function that describes what to do _after the component fully renders_ during `componentDidMount` and `componentDidUpdate`.
-1. A "dependency" array. This array contains references to any _props or state_ to _watch_. Whenever any watched props or state update, `useEffect` will run as part of `componentDidUpdate`.
-   - If we don't define this array, `useEffect` will be called during _every_ `componentDidUpdate`. That is, every time the component re-renders.
+1. A callback function that describes what to do _after the component fully renders_ just after the `Mounting Stage` or during `Update Stage`.
+1. A "dependency" array. This array contains references to any _props or state_ to _watch_. Whenever any watched props or state update, `useEffect` will run.
+   - If we don't define this array, `useEffect` will be called during _every_ `Update Stage`. That is, every time the component re-renders.
    - This array can be empty `[]` to indicate we are not watching for any updates.
+     - This means that the given callback function will only run once, when the component first renders.
```

Comment:
hmm i say we make this a call out since it's so important

```suggestion
### !callout-danger

## Watch Out for Infinite Loop

   - If we don't define this array, `useEffect` will be called during _every_ `Update Stage`. That is, every time the component re-renders. That can turn into an endless re-rendering loop.
   - This array can be empty `[]` to indicate we are not watching for any updates.
     - This means that the given callback function will only run once, when the component first renders.

### !end-callout
```